models/events/TestTakerImportListenerService.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018  (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoTestTaker\models\events;
+
+use core_kernel_classes_Resource;
+use oat\generis\Helper\UserHashForEncryption;
+use oat\generis\model\GenerisRdf;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\event\EventManagerAwareTrait;
+use oat\oatbox\log\LoggerAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use common_report_Report as Report;
+use oat\tao\model\event\CsvImportEvent;
+use oat\taoTestTaker\models\TestTakerSavePasswordInMemory;
+use oat\taoTestTaker\models\TestTakerService;
+
+class TestTakerImportListenerService extends ConfigurableService
+{
+    const SERVICE_ID = 'taoTestTaker/TestTakerImportListener';
+
+    use EventManagerAwareTrait;
+    use OntologyAwareTrait;
+    use LoggerAwareTrait;
+
+    /**
+     * @param CsvImportEvent $event
+     */
+    public function catchCsvImportEvent(CsvImportEvent $event)
```

Comment:
Why do u need this event to launch another? Could u do it directly do it the class/service itself?

Reply:
Because this is catching `CsvImportEvent` which is checking if it is a test taker resource, if yes it's dispatching a specific event for `TestTakerImport` which has other listeners to it. In our case the taoEncryption it's listing only to TestTakerImport event to handle the keys.